suave/sdk/sdk.go

```diff
@@ -64,29 +68,80 @@ func (c *Contract) SendTransaction(method string, args []interface{}, confidenti
 		return nil, err
 	}
 
-	computeRequest, err := types.SignTx(types.NewTx(&types.ConfidentialComputeRequest{
-		ConfidentialComputeRecord: types.ConfidentialComputeRecord{
-			KettleAddress: c.client.kettleAddress,
-			Nonce:         nonce,
-			To:            &c.addr,
-			Value:         nil,
-			GasPrice:      gasPrice,
-			Gas:           1000000,
-			Data:          calldata,
-		},
-		ConfidentialInputs: confidentialDataBytes,
-	}), signer, c.client.key)
+	record := &types.ConfidentialComputeRecord{
+		KettleAddress:          c.client.kettleAddress,
+		Nonce:                  nonce,
+		To:                     &c.addr,
+		Value:                  big.NewInt(0),
+		GasPrice:               gasPrice,
+		Gas:                    1000000,
+		Data:                   calldata,
+		ConfidentialInputsHash: crypto.Keccak256Hash(confidentialDataBytes), // ?
+
+		// initialize empty fields
+		ChainID: big.NewInt(1337), // suave chain id?
+		V:       big.NewInt(0),
+		R:       big.NewInt(0),
+		S:       big.NewInt(0),
+	}
+
+	/*
+		computeRequest, err := types.SignTx(types.NewTx(&types.ConfidentialComputeRequest{
+			ConfidentialComputeRecord: *record,
+			ConfidentialInputs:        confidentialDataBytes,
+		}), signer, c.client.key)
+		if err != nil {
+			return nil, err
+		}
+	*/
+
+	domain := &eip712.EIP712Domain{
+		Name:              "suave?",
+		Version:           "1.0",
+		VerifyingContract: "0x0000000000000000000000000000000000000000", // ??
```

Comment:
this is included in the eip712 spec because these signatures are typically validated by a smart contract; the contract would check that `msg.verifyingContract == address(this)`.
Since we're validating signatures offchain, I think this can be ignored and filled with a dummy value